src/apps/vpn/mozillavpn.cpp

```diff
@@ -1018,20 +1025,8 @@ void MozillaVPN::subscriptionStarted(const QString& productIdentifier) {
   logger.debug() << "Subscription started" << productIdentifier;
 
   setState(StateSubscriptionInProgress);
-
-  if (!Feature::get(Feature::Feature_webPurchase)->isSupported()) {
-    ProductsHandler* products = ProductsHandler::instance();
-
-    // If products are not ready (race condition), register the products again.
-    if (!products->hasProductsRegistered()) {
-      TaskScheduler::scheduleTask(new TaskProducts());
-      TaskScheduler::scheduleTask(new TaskFunction([this, productIdentifier]() {
-        subscriptionStarted(productIdentifier);
-      }));
-
-      return;
-    }
-  }
+  ProductsHandler* products = ProductsHandler::instance();
+  Q_ASSERT(products->hasProductsRegistered());
```

Comment:
This assertion is not valid anymore. Move this logic in ProductsHandler maybe in a `maybeStartSubscription(productIdentifier)`